text/convert.py

```diff
@@ -34,3 +37,125 @@ def run(self):
             for seq_tag, entry in d.items():
                 assert isinstance(entry, str), f"expected str, got {entry!r} (type {type(entry).__name__})"
                 out.write(entry + "\n")
+
+
+class TextDictToStmJob(Job):
+    """
+    Similar as :class:`CorpusToStmJob`, but does not use the Bliss XML, but instead the text dict as input
+    (e.g. via :class:`CorpusToTextDictJob`).
+    """
+
+    def __init__(
+        self,
+        text_dict: Path,
+        *,
+        exclude_non_speech: bool = True,
+        non_speech_tokens: Optional[List[str]] = None,
+        remove_punctuation: bool = True,
+        punctuation_tokens: Optional[Union[str, List[str]]] = None,
+        fix_whitespace: bool = True,
+        name: str = "",
+        tag_mapping: Sequence[Tuple[Tuple[str, str, str], Dict[int, Path]]] = (),
+    ):
+        """
+        :param text_dict: e.g. via :class:`CorpusToTextDictJob`
+        :param exclude_non_speech: non speech tokens should be removed
+        :param non_speech_tokens: defines the list of non speech tokens
+        :param remove_punctuation: should punctuation be removed
+        :param punctuation_tokens: defines list/string of punctuation tokens
+        :param fix_whitespace: should white space be fixed.
+            !!!be aware that the corpus loading already fixes white space!!!
```

Comment:
is this comment relevant for text dict or copy-pasted from the bliss job?
should we remove this parameter entirely if the `eval` call anyway corrects all whitespace errors?

Reply:
I copy-pasted the `CorpusToStmJob`. I personally don't use any of these options (i.e. I just use the defaults) and I did not really try to understand what they all do, i.e. which of those options are really necessary, and in what situations they make sense.

However, it seems to me that all of the options would apply to the text-dict file just in the same way as it would apply to the Bliss file, i.e. I don't see that there is anything specific about Bliss here, so I thought it makes sense to just have exactly the same logic.

Regarding this specific comment, I don't really understand what it means exactly, even for the `CorpusToStmJob`. I would assume it very much depends on your pipeline whether white space is already fixed. I don't know what "corpus loading" here refers to. I also don't exactly know what it means that "white space is fixed", i.e. what are the requirements for Sclite in the end. But again, whatever all this means, I don't think this is specific to Bliss, so this comment would apply to text-dict just as well, or not?

Reply:
The `eval` call would not touch the content of a string. If there are leading/trailing whitespaces (or other kind of whitespaces within?) the string, `eval` would not affect that. So this option here definitely has an effect.

Reply:
So, what to do here? Just remove the comment? Or just leave it?

Reply:
I guess they are referring to [this line](https://github.com/rwth-i6/i6_core/blob/de835b85a513a279a131d855ed12754643ae73e9/lib/corpus.py#L126) in the corpus lib, which has the same code as in this job.

Therefore if one converted Bliss corpora to stm, this option should be a redundant.

Here we don't have this, therefore I would say remove the comment and keep the option.
```suggestion
```